From 802089996e49bce2244a0faf1dbbce46e689451c Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:22 -0400
Subject: [cpufreq] pcc_freq: convert acpi_get_handle() to acpi_has_method()

Message-id: <1409235879-1283-49-git-send-email-prarit@redhat.com>
Patchwork-id: 89384
O-Subject: [RHEL7.1 PATCH BZ 1134639 048/185] pcc_freq: convert acpi_get_handle() to acpi_has_method()
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 7ca9b574915a17a588db37c5635890fe660ef7e9
Author: Zhang Rui <rui.zhang@intel.com>
Date:   Tue Sep 3 08:31:58 2013 +0800

    pcc_freq: convert acpi_get_handle() to acpi_has_method()

    acpi_has_method() is a new ACPI API introduced to check
    the existence of an ACPI control method.

    It can be used to replace acpi_get_handle() in the case that
    1. the calling function doesn't need the ACPI handle of the control method.
    and
    2. the calling function doesn't care the reason why the method is unavailable.

    Convert acpi_get_handle() to acpi_has_method()
    in drivers/cpufreq/pcc_freq.c in this patch.

    Signed-off-by: Zhang Rui <rui.zhang@intel.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/pcc-cpufreq.c b/drivers/cpufreq/pcc-cpufreq.c
index 70ce91b..f4e5dde 100644
--- a/drivers/cpufreq/pcc-cpufreq.c
+++ b/drivers/cpufreq/pcc-cpufreq.c
@@ -395,15 +395,14 @@ static int __init pcc_cpufreq_probe(void)
  struct pcc_memory_resource *mem_resource;
  struct pcc_register_resource *reg_resource;
  union acpi_object *out_obj, *member;
- acpi_handle handle, osc_handle, pcch_handle;
+ acpi_handle handle, osc_handle;
  int ret = 0;
 
  status = acpi_get_handle(NULL, "\\_SB", &handle);
  if (ACPI_FAILURE(status))
   return -ENODEV;
 
- status = acpi_get_handle(handle, "PCCH", &pcch_handle);
- if (ACPI_FAILURE(status))
+ if (!acpi_has_method(handle, "PCCH"))
   return -ENODEV;
 
  status = acpi_get_handle(handle, "_OSC", &osc_handle);
-- 
1.7.1